### PR TITLE
Fix azure e2e size test

### PR DESCRIPTION
**Pull request by lsviben**
**What does this PR do / Why do we need it**:
Tries to fix the azure e2e test:
TestAzureSizesWithCredentials
which got really flaky recently. It seems that the problem is that getting the Azure sizes sometimes lasts longer then 10s which is the default timeout. 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/test/e2e/utils/http.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 type Backoff wait.Backoff
 
 const (
-	apiRequestTimeout = 10 * time.Second
+	apiRequestTimeout = 30 * time.Second
```
Comment (xrstf) [inline pkg/test/e2e/utils/http.go:36]: Please leave a comment on this, mentioning Azure.

Comment (lsviben) [reply pkg/test/e2e/utils/http.go:36]: sure,

I can also revert to my first commit where I treat the azure test differently:
https://github.com/kubermatic/kubermatic/pull/10073/commits/180a6bae863dc3ca9a20aa063da942afdc7155f5

but I thought this is simpler. WDYT?

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**